gettsim/transfers/rente.py

```diff
@@ -295,4 +423,498 @@ def ges_rente_regelaltersgrenze(geburtsjahr: int, ges_rente_params: dict) -> flo
+        rates=ges_rente_params["altersgrenze_besonders_langjährig_versicherte"][
+            "rates"
+        ],
+        intercepts_at_lower_thresholds=ges_rente_params[
+            "altersgrenze_besonders_langjährig_versicherte"
+        ]["intercepts_at_lower_thresholds"],
+    )
+
+    return out
+
+
+def ges_rente_vorauss_vorzeitig(
+    ges_rente_vorrauss_frauen: bool,
+    ges_rente_vorrauss_langjährig: bool,
+) -> float:
+    """Function determining eligibility for early retirement.
+    Can only be claimed if eligible for "Rente für langjährig Versicherte"
+    or "Rente für Frauen" (or -not yet implemented - for disabled).
+
+    Parameters
+    ----------
+    ges_rente_vorrauss_frauen
+        See :func:`ges_rente_vorrauss_frauen`.
+    ges_rente_vorrauss_langjährig
+        See :func:`ges_rente_vorrauss_langjährig`.
+
+    Returns
+    -------
+    Eligibility as bool.
+
+    """
+
+    out = ges_rente_vorrauss_frauen or ges_rente_vorrauss_langjährig
+
+    return out
+
+
+def ges_rente_vorrauss_regelrente(ges_rente_wartezeit_5: float) -> bool:
+    """Function determining the eligibility for the Regelaltersrente.
+
+    Parameters
+    ----------
+    ges_rente_wartezeit_5
+        See :func:`ges_rente_wartezeit_5`.
+
+    Returns
+    -------
+    Eligibility as bool.
+
+    """
+    out = ges_rente_wartezeit_5 >= 5
+
+    return out
+
+
+def ges_rente_vorrauss_frauen(
+    weiblich: bool,
+    ges_rente_wartezeit_15: float,
+    y_pflichtbeitr_ab_40: float,
+    alter: int,
+    geburtsjahr: int,
+    ges_rente_params: dict,
+) -> bool:
+    """Function determining the eligibility for Altersrente für Frauen (pension
+    for women) Wartezeit 15 years, contributions 10 years after age 40,
+    being a women.
+
+    Parameters
+    ----------
+    weiblich
+        See basic input variable :ref:`weiblich <weiblich>`.
+    ges_rente_wartezeit_15
+        See :func:`ges_rente_wartezeit_15`
+    y_pflichtbeitr_ab_40
+        See basic input variable :ref:`y_pflichtbeitr_ab_40 <y_pflichtbeitr_ab_40>`.
+    alter
+        See basic input variable :ref:`alter <alter>`.
+    geburtsjahr
+        See basic input variable :ref:`geburtsjahr <geburtsjahr>`.
+    ges_rente_params
+        See params documentation :ref:`ges_rente_params <ges_rente_params>`.
+
+    Returns
+    -------
+    Eligibility as bool.
+
+    """
+    altersgrenze_vorzeitig = piecewise_polynomial(
+        x=geburtsjahr,
+        thresholds=ges_rente_params["altersgrenze_für_frauen_vorzeitig"]["thresholds"],
+        rates=ges_rente_params["altersgrenze_für_frauen_vorzeitig"]["rates"],
+        intercepts_at_lower_thresholds=ges_rente_params[
+            "altersgrenze_für_frauen_vorzeitig"
+        ]["intercepts_at_lower_thresholds"],
+    )
+
+    out = (
+        weiblich
+        and ges_rente_wartezeit_15 >= 15
+        and y_pflichtbeitr_ab_40 >= 10
+        and alter >= altersgrenze_vorzeitig
+    )
+
+    return out
+
+
+def ges_rente_vorrauss_langjährig(
```

Comment:
```suggestion
def ges_rente_vorauss_langj(
```

- please search and replace for "vorraus", affects many variables
- line up variable "besond_langj" with this one (add j)

Reply:
ok done.